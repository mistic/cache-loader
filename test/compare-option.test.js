const { webpack } = require('./helpers');

const mockCacheLoaderCompareFn = jest.fn();
const mockWebpackConfig = {
  loader: {
    options: {
      compare: () => {
        mockCacheLoaderCompareFn();
        return true;
      },
    },
  },
};

describe('compare option', () => {
  it('should call compare function', async () => {
    const testId = './basic/index.js';
    await webpack(testId, mockWebpackConfig);
    expect(mockCacheLoaderCompareFn).toHaveBeenCalled();
  });
});
