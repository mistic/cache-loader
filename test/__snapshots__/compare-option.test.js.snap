// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compare option should call compare with contextualized dep: errors 1`] = `Array []`;

exports[`compare option should call compare with contextualized dep: warnings 1`] = `Array []`;
